builds/jazz-build-module/apigee-module.groovy

```diff
@@ -0,0 +1,285 @@
+                } catch (e) {
+                    events.sendFailureEvent('APIGEE_API_PROXY_BUILD', e.getMessage(), apigeeContextMap)
+                    throw e
+                }
+
+                try {
+                    events.sendStartedEvent('APIGEE_API_PROXY_DEPLOY', 'Deploying Apigee API proxy bundle', apigeeContextMap)
+                    sh("./build/deploy.sh ${deployEnv.mgmt_host} ${deployEnv.mgmt_org} ${deployEnv.mgmt_env} ${templateValues.ProxyName} ${apiversion} ${USER} ${PASS} v1")
+                    events.sendCompletedEvent('APIGEE_API_PROXY_DEPLOY', 'Completed Apigee API proxy bundle deployment', apigeeContextMap)
+                } catch (e) {
+                    events.sendFailureEvent('APIGEE_API_PROXY_DEPLOY', e.getMessage(), apigeeContextMap)
+                    throw e
+                }
+            }
+        }
+        
+    }
+    
+    return hostUrl
+}
+
+/**
+ * Delete a previous deployed API proxy from Apigee
+ *
+ * @param swaggerFile The swagger file for the related API
+ * @param env_key The type of deployment. e.g. PROD or TEST.
+ * @param environment_logical_id The id used to differentiate proxies from different branches in the same domain.
+ * @param config The config metadata.
+ */
+def delete(swaggerFile, env_key, environment_logical_id, config) {
+    def templateValues = getTemplateValues(swaggerFile, environment_logical_id)    
+    def deployEnv = apigeeConfig.API_ENDPOINTS[env_key]
+    dir (apigeeModuleRoot) {
+        withCredentials([usernamePassword(credentialsId: apigeeConfig.apigeeCredId, passwordVariable: 'PASS', usernameVariable: 'USER')]) {
+            try {
+                events.sendStartedEvent('APIGEE_API_PROXY_DELETE', 'Deleting Apigee API proxy.')
+                sh("./build/delete.sh ${deployEnv.mgmt_host} ${deployEnv.mgmt_org} ${deployEnv.mgmt_env} ${templateValues.ProxyName} ${USER} ${PASS}")
+                events.sendCompletedEvent('APIGEE_API_PROXY_DELETE', 'Completed deletion of Apigee API proxy.')
+            } catch (e) {
+                events.sendFailureEvent('APIGEE_API_PROXY_DELETE', e.getMessage())
+                throw e
+            }
+        }
+    }
+}
+
+/**
+ * Run the proxygen process to generate the api proxy definition files.
+ *
+ * @param templateValues A Map of token name, replacement values to use when filling templates.
+ */
+def proxygen(templateValues) {
+    dir ('gen') {
+        deleteDir()
+    }    
+    dir ('templates/CoreAPI') {
+        deleteDir()
+    }
+    dir ('templates') {
+        sh "mkdir ./CoreAPI"
+        if (apigeeConfig.useSecure) {
+            sh ("cp -r ./coreAPI_secure/* ./CoreAPI")
+        } else {
+            sh ("cp -r ./coreAPI_default/* ./CoreAPI")
+        }
+    }
+
+    applyTemplate('proxygen_template.properties', '.', 'proxygen.properties', templateValues)
+    sh ('cat proxygen.properties')
+    sh ("java -cp ./ProxyGen.jar com.tmobile.apigee.ProxyCodeGen proxygen.properties")
```

Comment:
should this be called tmobile @devsatishm ?

Reply:
This is part of the namespace when the jar was compiled. Hence should be fine.